primap2/_downscale.py

```diff
@@ -158,6 +159,59 @@ def downscale_timeseries(
 
         return self._da.fillna(downscaled)
 
+    def downscale_timeseries_by_shares(
+        self,
+        *,
+        dim: Hashable,
+        basket: Hashable,
+        basket_contents: Sequence[Hashable],
+        basket_contents_shares: xr.DataArray,
+    ) -> xr.DataArray:
+        """Downscale timeseries along a dimension using defined shares for each timestep.
+
+        This is useful if you have data points for a total, and you don't have any
+        data for the higher resolution, but you do have the shares for the higher
+        resolution from another source. For example, you have the total energy, and
+        you know the shares of the sub-sectors 1.A and 1.B for each year from another
+        source.
+
+        Parameters
+        ----------
+        dim : Hashable
+            The dimension along which to perform the downscaling (e.g., "category" or "area").
+        basket : Hashable
+            The label of the aggregate group (e.g., "1.A" for a category) whose value will be
+            redistributed.
+        basket_contents : Sequence of Hashable
+            The labels of the subgroups (e.g., ["1.A.1", "1.A.2", "1.A.3"])
+            that make up the `basket`.
+        basket_contents_shares : xr.DataArray
+            The shares to use for downscaling.
+
+        Returns
+        -------
+        xr.DataArray
+            A new datarray with variables downscaled along `dim` using the provided shares.
+        """
+
+        # normalise shares
```

Comment:
```suggestion
        # normalise shares
        basket_contents_shares = basket_contents_shares.pr.loc[{dim: basket_contents}]
```
I think it makes sense to first select only the actual basket contents. If for example the reference data contain "1.A.1", "1.A.2" and "1.A.3", but basket_contents are only "1.A.1" and "1.A.2", then you would otherwise later not preserve the value of "1.A" fully.